src/Ko/SharedMemory.php

```diff
@@ -185,6 +208,45 @@ public function offsetUnset($offset)
      */
     public function count()
     {
-        return count($this->keyMapper);
+        $task = function() {
+            return count($this->getKeyMap());
+        } ;
+        
+        return $this->lockExecute($task) ;
+    }
+    
+    /**
+     * Lock shared memory if lock not already acquired by this process
+     * 
+     * @param function $task Function to execute
+     * 
+     * @return mixed    Whatever the $task function returns
+     */
+    protected function lockExecute($task)
+    {
+        if($this->mutex->isAcquired())
```

Comment:
Please use short return style and PSR-0;

if (cond) {
  return ...
}

return ...